cmd/csi_driver/main.go

```diff
@@ -42,13 +55,203 @@ var (
+	}
+	klog.Infof("Found gateway %s for %s", gateway.String(), nicName)
+
+	// Define and add the route
+	_, dst, _ := net.ParseCIDR("10.170.0.0/16") // MGS IP
+	route := &netlink.Route{
+		LinkIndex: link.Attrs().Index,
+		Dst:       dst,
+		Gw:        gateway,
+		Table:     tableID,
+	}
+	if err := netlink.RouteReplace(route); err != nil {
+		return fmt.Errorf("failed to replace/add route for %s: %w", nicName, err)
+	}
+	klog.Infof("Successfully added (or replace) route for %s", nicName)
+
+	// Define and add the rule
+	rule := netlink.NewRule()
+	rule.Table = tableID
+	rule.Src = &net.IPNet{IP: ipAddr, Mask: net.CIDRMask(32, 32)} // /32 mask for a single IP
+	if err := netlink.RuleAdd(rule); err != nil {
+		return fmt.Errorf("failed to add rule for %s: %w", nicName, err)
+	}
+	klog.Infof("Successfully added rule for %s", nicName)
+
+	return nil
+}
+
+func isMultiRailEnabled() (bool, error) {
+	// TODO(halimsam): Add Multi Rail Logic here.
+	return false, nil
+}
+
+func getLnetPort() int {
+	legacyPortEnabled := os.Getenv("ENABLE_LEGACY_LUSTRE_PORT")
+	if legacyPortEnabled == "true" {
+		return 6988
+	}
+
+	return 988
+}
+
+func checkLnetPort(expectedPort int) error {
+	file, err := os.ReadFile(lnetPortFile)
+	if err != nil {
+		if os.IsNotExist(err) {
+			klog.V(4).Infof("LNET port file %v not found. Skipping check", lnetPortFile)
+
+			return nil
+		}
+
+		return fmt.Errorf("error reading file %s: %w", lnetPortFile, err)
+	}
+	currPort := strings.TrimSpace(string(file))
+	expectedPortStr := strconv.Itoa(expectedPort)
+	if currPort != expectedPortStr {
+		klog.V(3).Infof("Your node already has Lustre kernel modules installed with an outdated configuration. Please upgrade your node pool to apply the correct settings.")
+
+		return fmt.Errorf("lnet port mismatched, expected %s, but found: %s", expectedPortStr, currPort)
+	}
+
+	log.Println("LNET_PORT matches configuration. Check passed.")
+
+	return nil
+}
+
+func configureRoute(nicName string, tableID int) error {
+	link, err := netlink.LinkByName(nicName)
+	if err != nil {
+		return fmt.Errorf("could not get link for %s: %w", nicName, err)
+	}
+	addrs, err := netlink.AddrList(link, netlink.FAMILY_V4)
+	if err != nil || len(addrs) == 0 {
+		return fmt.Errorf("could not get address for %s: %w", nicName, err)
+	}
+	klog.Infof("IP address are: %+v", addrs)
+	ipAddr := addrs[0].IP
+	if err := configureRoutesForNic(nicName, ipAddr, tableID); err != nil {
+		return fmt.Errorf("failed to configure routes for %s: %w", nicName, err)
+	}
+
+	return nil
+}
+
+func kmodInstaller() error {
+	if err := checkLnetPort(getLnetPort()); err != nil {
+		return err
+	}
+	nics, err := getGvnicNames()
+	if err != nil {
+		return fmt.Errorf("error getting nic names: %w", err)
+	}
+	if len(nics) == 0 {
+		return fmt.Errorf("no nics with eth prefix found")
+	}
+	isMultiNic, err := isMultiRailEnabled()
+	if err != nil {
+		return err
+	}
+	klog.V(4).Infof("Is using LustreCSI Multi-Rail feature: %v", isMultiNic)
```

Comment:
Let's make sure to polish the logs for multi NIC related functions in the follow-up PR